### PR TITLE
Add commands for listing roles and members assigned a role in an organization

### PR DESCRIPTION
**Pull request by sergeybykov**
### Description

This adds two commands.

`auth0 orgs roles list <org ID>` for listing roles that are assigned to members of an organization.
and
`auth0 orgs roles members list <org ID> <role ID>` for listing members of an organization that have a given role assigned.

The current implementation is not efficient because it has to enumerate all members and make a call for each member to retrieve their roles. There is an [outstanding ask](https://community.auth0.com/t/find-members-from-organization-with-specific-role/65626/10?u=sbykov) to Auth0 to add APIs for getting roles and members with a role assigned with single API calls instead of enumerating members.

### References

#379
[Forum thread](https://community.auth0.com/t/find-members-from-organization-with-specific-role/65626/10?u=sbykov) 

### Testing

Manual testing against a tenant with an organization populated with members that have roles assigned to them.

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [X] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [X] All active GitHub checks for tests, formatting, and security are passing
- [X] The correct base branch is being used, if not `master`

### CODE REVIEW

File: internal/auth0/organization.go [hunk 1]
```diff
@@ -32,4 +32,9 @@ type OrganizationAPI interface {
 	//
 	// See: https://auth0.com/docs/api/management/v2#!/Organizations/get_members
 	Members(id string, opts ...management.RequestOption) (o *management.OrganizationMemberList, err error)
+
+	// List roles assigned to a member of an organization
```
Comment (Widcket) [inline internal/auth0/organization.go:36]:
```suggestion
	// MemberRoles lists roles assigned to a member of an organization
```

File: internal/cli/organizations.go [hunk 2]
```diff
@@ -461,33 +464,140 @@ auth0 orgs members ls <id>`,
...
-					if apiErr != nil {
-						return nil, false, apiErr
-					}
-					var output []interface{}
-					for _, member := range members.Members {
-						output = append(output, member)
-					}
-					return output, members.HasNext(), nil
-				})
+			members, err := cli.getOrgMembers(cmd.Context(), inputs.ID, inputs.Number)
```
Comment (Widcket) [inline internal/cli/organizations.go:467]: The async operations aren't being done inside a spinner block.

File: internal/cli/organizations.go [hunk 3]
```diff
@@ -461,33 +464,140 @@ auth0 orgs members ls <id>`,
...
+			if len(args) == 0 {
+				err := organizationID.Pick(cmd, &inputs.ID, cli.organizationPickerOptions)
+				if err != nil {
+					return err
+				}
+			} else {
+				inputs.ID = args[0]
+			}
+
+			members, err := cli.getOrgMembers(cmd.Context(), inputs.ID, inputs.Number)
```
Comment (Widcket) [inline internal/cli/organizations.go:517]: Same as above: the async operations aren't being done inside a spinner block.

File: internal/cli/organizations.go [hunk 4]
```diff
@@ -461,33 +464,140 @@ auth0 orgs members ls <id>`,
...
+				return err
 			}
-			var typedList []management.OrganizationMember
-			for _, item := range list {
-				typedList = append(typedList, item.(management.OrganizationMember))
+
+			roleMap := make(map[string]management.OrganizationMemberRole)
+			for _, member := range members {
+				userID := member.GetUserID()
+				roleList, err := cli.api.Organization.MemberRoles(inputs.ID, userID)
```
Comment (Widcket) [inline internal/cli/organizations.go:525]: All this logic (that does network calls inside a given command) should be inside a single spinner block.

File: internal/cli/organizations.go [hunk 5]
```diff
@@ -461,33 +464,140 @@ auth0 orgs members ls <id>`,
...
+				if err != nil {
+					return err
+				}
+				inputs.RoleID = args[0]
+			} else {
+				inputs.OrgID = args[0]
+				inputs.RoleID = args[1]
+			}
+
+			members, err := cli.getOrgMembers(cmd.Context(), inputs.OrgID, inputs.Number)
```
Comment (Widcket) [inline internal/cli/organizations.go:577]: Same as above.

File: internal/cli/organizations.go [hunk 6]
```diff
@@ -461,33 +464,140 @@ auth0 orgs members ls <id>`,
...
+			if err != nil {
+				return err
+			}
 
+			var roleMembers []management.OrganizationMember
+			for _, member := range members {
+				userID := member.GetUserID()
+				roleList, err := cli.api.Organization.MemberRoles(inputs.OrgID, userID)
```
Comment (Widcket) [inline internal/cli/organizations.go:585]: Same as above.

File: internal/cli/organizations.go [hunk 7]
```diff
@@ -461,33 +464,140 @@ auth0 orgs members ls <id>`,
...
+		Number int
+	}
+
+	cmd := &cobra.Command{
+		Use:   "members",
+		Args:  cobra.RangeArgs(1, 2),
+		Short: "List organization members for a role",
+		Long:  "List organization members that have a given role assigned to them.",
+		Example: `auth0 orgs roles members
+auth0 orgs roles members <org id> <role id>`,
```
Comment (Widcket) [inline internal/cli/organizations.go:564]: Please, have this command be `auth0 orgs roles members list <org-id> --role-id role`. See `--api-id` in https://auth0.github.io/auth0-cli/auth0_roles_permissions_add.html as an example.
`--role-id` (`-r`) should be a required flag.

**Review by Widcket — CHANGES_REQUESTED** [review]
Thanks for raising this PR.

Please have the `auth0 orgs roles members <org id> <role id>` command be `auth0 orgs roles members list <org-id> --role-id role` instead, and use spinners for all async operations (one spinner start/stop in the body of a given command).

**Comment by sergeybykov** [comment]
@Widcket Thanks for the feedback. I think I addressed it all. Could you please take a look.

**Review by Widcket — CHANGES_REQUESTED** [review]
Running `auth0 orgs roles members list`, it didn't ask to pick the role ID. See https://github.com/auth0/auth0-cli/blob/main/internal/cli/roles_permissions.go#L116 for an example.

**Comment by sergeybykov** [comment]
@Widcket I added a role picker. Note though that it shows *all* roles within the tenant, not just those used assigned to members of the org. The latter would be cleaner but I think the user experience would be much worse because it takes a long time and many roundtrips to build such a list. We have to do that eventually anyway, but at least that isn't blocking user input.

**Comment by sergeybykov** [comment]
Added a spinner for the loop that retrieves roles assigned to each member.

**Comment by Widcket** [comment]
@sergeybykov thanks for making the changes! I'm OOO at the moment, so will re-review when I get back.

File: internal/cli/organizations.go [hunk 8]
```diff
@@ -461,38 +472,175 @@ auth0 orgs members ls <id>`,
...
+						if _, exists := roleMap[roleID]; !exists {
+							roleMap[roleID] = role
+						}
 					}
-					return output, members.HasNext(), nil
-				})
-
+				}
+				return nil
+			})
```
Comment (sergiughf) [inline internal/cli/organizations.go:546]: Would it be possible to abstract this within a func?:) Similar to what we did for `cli.getOrgMembersWithSpinner()` above?

e.g.
```go
memberRoleMap, err := cli.getOrgMemberRolesWithSpinner(...)
```

File: internal/cli/organizations.go [hunk 9]
```diff
@@ -461,38 +472,175 @@ auth0 orgs members ls <id>`,
...
-				typedList = append(typedList, item.(management.OrganizationMember))
+			// convert management.OrganizationMemberRole to management.Role
+			var roles []*management.Role
+			for _, role := range roleMap {
+				roles = append(roles, &management.Role{ID: role.ID, Name: role.Name, Description: role.Description})
 			}
-			cli.renderer.MembersList(typedList)
+			sort.Slice(roles, func(i, j int) bool {
+				return strings.ToLower(roles[i].GetName()) < strings.ToLower(roles[j].GetName())
+			})
```
Comment (sergiughf) [inline internal/cli/organizations.go:557]: Same thing as above, would be great to move this within a func such that we don't need to add the `convert` comment but instead the func name would be self explanatory e.g. `convertOrganizationMemberRolesToRoles()`.

File: internal/cli/organizations.go [hunk 10]
```diff
@@ -461,38 +472,175 @@ auth0 orgs members ls <id>`,
...
+							roleMembers = append(roleMembers, member)
+						}
+					}
+				}
+				return nil
+			})
+			if errSpinner != nil {
+				return errSpinner
+			}
+			sortMembers(roleMembers)
```
Comment (sergiughf) [inline internal/cli/organizations.go:633]: This looks like a candidate as well to be moved within a func so it's easier to navigate the codebase. Wdyt?

**Review by sergiughf — COMMENTED** [review]
Thanks for your contribution @sergeybykov! I cloned this locally and everything seems to work as per @Widcket's previous feedback. 

I only left a couple of small comments regarding refactoring the code a little bit so it's easier to navigate.

**Comment by sergeybykov** [comment]
@sergiughf I believe I addressed all your comments.

**Review by sergiughf — APPROVED** [review]
Thank you for your contribution @sergeybykov! 🥳  Really good work put into this PR.

**Pull request merged**